### PR TITLE
Improve experience for NU3004 and NU3034

### PR DESCRIPTION
**Pull request by PatoBeltran**
## Bug
Fixes: https://github.com/NuGet/Home/issues/6906

## Fix
Details: Added more detailed tests for the scenarios where unsigned packages are not allowed and for allow list issues (either empty or no match for require mode and repository announcing all packages signed.)

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/Strings.resx [hunk 1]
```diff
@@ -590,13 +590,13 @@ Valid from:</comment>
     <value>The primary signature and repository countersignature are unrelated.</value>
   </data>
   <data name="Error_NoRepoAllowList" xml:space="preserve">
-    <value>A repository announced that their packages should be signed but an empty list of trusted certificates was found.</value>
+    <value>The repository where this package was downloaded announced that their packages should be signed but an empty list of trusted signers was found. No package downloaded from this repository can be trusted.</value>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging/Strings.resx]: The repository ~~where~~ this package was downloaded **from** maybe?

The where is superfluous.

Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx]: Do we really want the `No package downloaded from this repository can be trusted`
Isn't this the case where you are not trusting any repositories?

File: src/NuGet.Core/NuGet.Packaging/Strings.resx [hunk 2]
```diff
@@ -786,4 +786,10 @@ Valid from:</comment>
   <data name="NuGetLicense_LicenseElementMissingValue" xml:space="preserve">
     <value>The license element value is empty. This is likely due to an authoring error. </value>
   </data>
+  <data name="Error_RepositorySettings_UnsignedPackage" xml:space="preserve">
+    <value>The repository announced that all their packages where signed and the package was found unsigned.</value>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging/Strings.resx]: The repository announced that all their packages where signed and the package was found unsigned. to
The repository announced that all their packages **are** signed but the package {0} is not.

Do we have information about the package id here? 
We should add it. 
If there's a wrapper over this error that specifies that, simply removing `{0}` should be enough. 

Happy to discuss these message offline.

**Review by nkolev92 — COMMENTED** [review]
Some questions about the messages. 
Definitely looks like the right direction.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx (hunk 1 above)
Comment (PatoBeltran) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx]: This is the case where a package said that all their packages are signed, but didn't give a list of trusted signers. I can remove that part of the message, I just wanted to add something that clearly states to the user why we are failing, since they don't have control over what the server is sending.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx (hunk 2 above)
Comment (PatoBeltran) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx]: This will be wrapped in another message containing package id, version and source. So the full message will be: 

```
Package '{0} {1}' from source '{2}': The repository announced that all their packages are signed, but the package is not.
```

File: src/NuGet.Core/NuGet.Packaging/Strings.resx (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx]: I see. 
It's fine then, I just thought that the last part was not clear enough. 

Not sure if a customer will understand it.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx (hunk 2 above)
Comment (nkolev92) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx]: Sounds good.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx (hunk 1 above)
Comment (PatoBeltran) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx]: Any suggestion on how to improve this so it's easier for the customer to understand?

Comment (PatoBeltran) [reply src/NuGet.Core/NuGet.Packaging/Strings.resx]: `The repository this package was downloaded from announced that their packages are signed but an empty list of trusted signers was found. This is likely an issue with the repository.`

File: src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs [hunk 3]
```diff
@@ -1121,8 +1121,13 @@ private static void DeleteTargetAndTempPaths(string targetPath, string targetTem
         /// <param name="source">package source.</param>
         /// <param name="package">package identity.</param>
         /// <param name="message">ILogMessage to be modified.</param>
-        private static void AddPackageIdentityToSignatureLog(string source, PackageIdentity package, SignatureLog message)
+        private static void UpdateSignatureLog(string source, PackageIdentity package, ClientPolicyContext policyContext, SignatureLog message)
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/PackageExtractor.cs:1118]: `policyContext` is missing from the XML doc comments.  Instead of adding it though, I'd just remove the XML doc comments entirely; it's a private method.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx [hunk 4]
```diff
@@ -519,7 +519,7 @@ Valid from:</comment>
     <value>The argument is invalid.</value>
   </data>
   <data name="Error_NoMatchingClientCertificate" xml:space="preserve">
-    <value>The package signature certificate fingerprint does not match any certificate fingerprint in client allow list.</value>
+    <value>The package signature certificate fingerprint does not match any certificate fingerprint from the user's trusted signers. You need to trust the signer.</value>
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Strings.resx]: This package is signed but not by a trusted signer.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx [hunk 5]
```diff
@@ -590,13 +590,13 @@ Valid from:</comment>
     <value>The primary signature and repository countersignature are unrelated.</value>
   </data>
   <data name="Error_NoRepoAllowList" xml:space="preserve">
-    <value>A repository announced that their packages should be signed but an empty list of trusted certificates was found.</value>
+    <value>The repository this package was downloaded from announced that their packages are signed but an empty list of trusted signers was found. This is likely an issue with the repository.</value>
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Strings.resx]: This repository indicated that all its packages are repository signed; however, it listed no signing certificates.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx [hunk 6]
```diff
@@ -590,13 +590,13 @@ Valid from:</comment>
...
   </data>
   <data name="Error_NoClientAllowList" xml:space="preserve">
-    <value>A list of trusted signers is required by the client but none was found.</value>
+    <value>If in require mode, the user must to have a list of trusted signers but none was found. Please add trusted signers to the configuration file.</value>
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Strings.resx]: signatureValidationMode is set to require, so packages are allowed only if signed by trusted signers; however, no trusted signers were specified.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx [hunk 7]
```diff
@@ -590,13 +590,13 @@ Valid from:</comment>
...
   </data>
   <data name="Error_NoMatchingRepositoryCertificate" xml:space="preserve">
-    <value>The package signature certificate fingerprint does not match any certificate fingerprint in repository allow list.</value>
+    <value>The package signature certificate fingerprint does not match any trusted signer given from the repository where the package was downloaded from. This package is untrusted.</value>
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Strings.resx]: This package was not repository signed with a certificate listed by this repository.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx [hunk 8]
```diff
@@ -786,4 +786,10 @@ Valid from:</comment>
   <data name="NuGetLicense_LicenseElementMissingValue" xml:space="preserve">
     <value>The license element value is empty. This is likely due to an authoring error. </value>
   </data>
+  <data name="Error_RepositorySettings_UnsignedPackage" xml:space="preserve">
+    <value>The repository announced that all their packages are signed, but the package is not.</value>
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Strings.resx]: This repository indicated that all its packages are repository signed; however, this package is unsigned.

File: src/NuGet.Core/NuGet.Packaging/Strings.resx [hunk 9]
```diff
@@ -786,4 +786,10 @@ Valid from:</comment>
...
+  </data>
+  <data name="Error_RequireMode_UnsignedPackage" xml:space="preserve">
+    <value>The package is unsigned and require mode does not allow installation of unsigned packages.</value>
```
Comment (dtivel) [inline src/NuGet.Core/NuGet.Packaging/Strings.resx]: signatureValidationMode is set to require, so packages are allowed only if signed by trusted signers; however, this package is unsigned.

**Review by dtivel — APPROVED** [review]
